src/main/scala/cognite/spark/v1/PushdownUtilities.scala

```diff
@@ -154,15 +162,11 @@ object PushdownUtilities {
     }
 
   def externalIdsSeqFromWrappedArray(wrappedArray: String): Seq[String] =
-    // We get "WrappedArray(ext1, ext2)" here, so we can remove irrelevant parts of the string
-    // and extract ext1 and ext2 from it -> return Seq('ext1', 'ext2')
-    wrappedArray.length match {
-      case 14 => Seq() // "WrappedArray()".length
-      case _ => wrappedArray.slice(13, wrappedArray.length - 1).split(',')
-    }
+    // we get a string separated by \0
+    wrappedArray.split('\u0000')
```

Comment:
Does that mean it's not a `WrappedArray` anymore, since we don't bother with removing the irrelevant parts of the string anymore? If so, should the name be changed?

Reply:
Cool, but then what will this look like when running on 2.12? :thinking:

Reply:
For sure, this works on 2.12. It used to be WrappedArray, because we just called `.toString` on anything that was passed to the pushdown filter logic. Now I explicitly use the null-byte separator, so it behaves in the same way on both Scala versions.